### PR TITLE
Feat ERF-61 - Search page

### PR DESCRIPTION
**Pull request by vogelino**
This PR adds the Search page with place-suggestions and search results.

### CODE REVIEW

File: pages/search.tsx [hunk 1]
```diff
@@ -11,13 +28,148 @@ export const getServerSideProps: GetServerSideProps = async ({ query }) => ({
...
+    <span className="text-gray-300 transform -translate-y-2">
+      <GeoPinIcon />
+    </span>
+    <div
+      className={classNames(
+        'pb-2 border-b border-gray-300 flex-grow border-dashed',
+        'group-hover:border-opacity-0 group-focus:border-opacity-0 transition'
+      )}
+    >
+      <h6 className="text-sm leading-4">
```
Comment (dnsos) [inline pages/search.tsx:95]: I think that somewhere around here happens what leads to this:

Screenshot 2021-06-22 at 13 39 26

Is there any way that we can make this case-insensitive? Or if that is too tricky, I wouldn't mind if we just remove the gray highlighting altogether.

File: pages/search.tsx [hunk 2]
```diff
@@ -11,13 +28,148 @@ export const getServerSideProps: GetServerSideProps = async ({ query }) => ({
...
+  return (
+    <div>
+      <h4 className="font-bold text-xl hidden sm:block">Standort</h4>
+      <p className="text-sm text-gray-500 hidden sm:block">
+        Finde deinen Bezirk, deine Straße oder einen anderen Ort in Berlin.
+      </p>
+
+      <input
+        type="text"
+        placeholder="Geben Sie einen Ort an"
```
Comment (dnsos) [inline pages/search.tsx:132]: Let's make this less formal: "Gib hier einen Ort an"

File: src/components/Map/index.tsx [hunk 3]
```diff
@@ -26,16 +32,44 @@ export const Map: FC<MapProps> = ({
   children,
   ...otherMapProps
 }) => {
+  const { pathname, query, replace } = useRouter()
+  const mappedQuery = mapRawQueryToState(query)
+  const transitionProps = {
+    transitionDuration: 2000,
+    transitionEasing: (t: number) =>
+      t < 0.5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1,
```
Comment (dnsos) [inline src/components/Map/index.tsx:40]: Wow, what is this magic? 😄

File: src/modules/RefreshmentMap/content.ts [hunk 4]
```diff
@@ -226,3 +241,202 @@ export const ABOUT_ACCORDION_ITEMS = [
...
+    },
+    geometry: {
+      type: 'Point',
+      coordinates: [13.462956678190112, 52.535800958365236],
+    },
+  },
+  {
+    type: 'Feature',
+    properties: {
+      name: 'Wasserfall am Victoriapark',
```
Comment (dnsos) [inline src/modules/RefreshmentMap/content.ts:263]: "Viktoriapark" mit a "k"

File: src/modules/RefreshmentMap/index.tsx [hunk 5]
```diff
@@ -181,7 +181,7 @@ export const RefreshmentMap: FC<RefreshmentMapPropType> = (pageProps) => {
           <Sidebar {...pageProps} />
           <div
             className={classNames(
-              'absolute transform z-50',
+              'absolute transform z-50 pointer-events-none',
```
Comment (dnsos) [inline src/modules/RefreshmentMap/index.tsx:185]: This seems to make the `HourSelector` inaccessible by mouse click. Can you double-check that?

What is actually the reason to make the pointer-events `none`?

**Review by dnsos — CHANGES_REQUESTED** [review]
Great work @vogelino! I added a few comments with some minor suggestions/questions.

I noticed that the query parameters do not bring you to the desired location when opening a URL initially, but I see that you added another commit in the meantime that might fix this?

File: pages/search.tsx (hunk 1 above)
Comment (vogelino) [reply pages/search.tsx:95]: Good point! I think I know a way. I'll try that out! Thx!

File: pages/search.tsx (hunk 2 above)
Comment (vogelino) [reply pages/search.tsx:132]: Will do!

File: src/components/Map/index.tsx (hunk 3 above)
Comment (vogelino) [reply src/components/Map/index.tsx:40]: Hahaha it is a copy/pasted easeInOutQuad function. I will extract it and give it a name. :D

File: src/modules/RefreshmentMap/content.ts (hunk 4 above)
Comment (vogelino) [reply src/modules/RefreshmentMap/content.ts:263]: Ok. This comes right from Lisa's GeoJSON. I'll fix that!

File: src/modules/RefreshmentMap/index.tsx (hunk 5 above)
Comment (vogelino) [reply src/modules/RefreshmentMap/index.tsx:185]: This is on purpose as it prevents the scroll of sidebars and the panning of the map to happening. If you look at the interactive children of HourSelector, they set `pointer-events-auto` to counter this on a more local level.

**Comment by vogelino** [comment]
Thx @dnsos for the thorough review! I'll make changes and let you know!

> One more idea: maybe we can zoom in a bit more to a selected location?

Will do!

> I noticed that the query parameters do not bring you to the desired location when opening a URL initially, but I see that you added another commit in the meantime that might fix this?

Exactly, the last commit fixes this issue!

**Comment by vogelino** [comment]
@dnsos I made corrections based on your comments! 👌  There were all excellent points! Can you have another look!? Cheers! 🤜 🤛

**Review by lucasoeth — APPROVED** [review]
Amazing work yo! Everything looks good to me. It seems like you have fixed all of @dnsos comments. Feel free to merge it !

**Comment by vogelino** [comment]
> rearrange the SEARCH_SUGGESTIONS array manually

I agree! However, I know that lisa will convert the content.ts file into a tsx file which will complicate the merging process. I'll do that later on when she is done with her changes and merge this branch for now.

**Review by dnsos — APPROVED** [review]
Thanks for taking the suggestions into consideration, @vogelino. Very good fixes and improvements in the last commits!

**Pull request merged**